core/consensus/grandpa/vote_graph/vote_graph_impl.cpp

```diff
@@ -281,19 +283,18 @@ namespace kagome::consensus::grandpa {
       return std::nullopt;
     }
 
-    std::deque<std::reference_wrapper<const Entry>> observing_entries;
-
-    auto start_node = active_node;
-    observing_entries.emplace_back(start_node);
+    /// entries to be processed
+    std::stack<std::reference_wrapper<const Entry>> nodes;
 
-    while (not observing_entries.empty()) {
-      auto &entry = observing_entries.front().get();
-      observing_entries.pop_front();
+    nodes.push(active_node);
+    while (not nodes.empty()) {
+      auto &node = nodes.top().get();
+      nodes.pop();
```

Comment:
Hm... **pop()** makes gotten **node** dangling reference.
Please, move using of **pop()** after using of **node**

Reply:
to be discussed